.circleci/config.yml

```
@@ -1,41 +1,162 @@
-version: 2
+version: 2.1
+
+defaults: &defaults
+  environment: &environment
+    CIRCLE_TEST_REPORTS: /tmp/test-results
+    CIRCLE_ARTIFACTS: /tmp/test-artifacts
+    BUNDLE_JOBS: 4
+    BUNDLE_RETRY: 3
+    BUNDLE_PATH: ~/spree/vendor/bundle
+  working_directory: ~/spree
+  docker:
+    - image: &ruby_image circleci/ruby:2.7-node-browsers
+    - image: &redis_image circleci/redis:6.2-alpine
+
+defaults_3_0: &defaults_3_0
+  <<: *defaults
+  docker:
+    - image: &ruby_3_0_image circleci/ruby:3.0-node-browsers
+    - image: *redis_image
+
+run_tests: &run_tests
+  <<: *defaults
+  parallelism: 8
```

```suggestion
  parallelism: 2
```

more than 2 doesn't make sense for this repo